node/pkg/processor/observation.go

```diff
@@ -318,12 +319,24 @@ func (p *Processor) handleInboundSignedVAAWithQuorum(ctx context.Context, m *gos
 	//  - enough signatures are present for the VAA to reach quorum
 
 	// Check if we already store this VAA
-	_, err = p.db.GetSignedVAABytes(*db.VaaIDFromVAA(v))
+	b, err := p.db.GetSignedVAABytes(*db.VaaIDFromVAA(v))
 	if err == nil {
-		p.logger.Debug("ignored SignedVAAWithQuorum message for VAA we already store",
-			zap.String("digest", hash),
-		)
-		return
+		existing, err := vaa.Unmarshal(b)
+		if err != nil {
+			// Undefined state: This can't happen unless the DB is corrupted.
+			panic(fmt.Sprintf("failed to unmarshal VAA: %v", err))
+		}
+
+		// Accept new VAA if the new one has a different guardian set index.
+		//
+		// This allows messages re-signed by a different guardian set to be persisted by nodes
+		// even on nodes that failed to re-observe the message.
+		if existing.GuardianSetIndex == v.GuardianSetIndex {
```

Comment:
This is sort of a rhetorical question, but should the logic to re-sign only happen if the different guardian set index is higher than the existing one? I'm trying to think if it is possible for there to be hijinks with someone trying to use a lower index and think the answer is no, but just wanted to think about it.

Reply:
This is a potential DoS vector, say you have a signed VAA by guardian set N, then there's an upgrade to N+1 and a reobservation request for the VAA. Now, as long as a guardian holds on to the VAA signed by N, they can keep broadcasting it, kicking out the N+1 VAA from storage. Pretty unlikely event, but theoretically possible, so we should only override with newer guardian sets.